### ISSUE TITLE
Bounding box for label does not set focus

### ISSUE THREAD

**Issue by RivenSkaye** [AvaloniaUI/Avalonia#17701]
### Describe the bug

When adding a label and another element to a document tree, the label's bounding box does not trigger focus onto the other element _unless_ its background property has been set. The label's inner text _does_ trigger the focus change regardless.

### To Reproduce

Shove this into a Window and click anywhere in the bounding box of either label, but not on the text itself. Setting the background to `Transparent` works, but leaving it out does not.

Violet is for illustrative purposes - it visualizes the bounding box size
```xaml
<Grid Width="{Binding $parent.Bounds.Width}" ColumnDefinitions="Auto, 120, Auto, 120"
          HorizontalAlignment="Center">
      <Label VerticalAlignment="Center" VerticalContentAlignment="Center" HorizontalAlignment="Center"
             Content="Clicking outside the text does not change focus"
             Height="{Binding $parent.Bounds.Height}" Target="FactInput"
             HorizontalContentAlignment="Center" Grid.Column="0" />
      <TextBox Grid.Column="1" Watermark="I don't get focus" Name="FactInput" />
      <Label VerticalAlignment="Center" VerticalContentAlignment="Center" HorizontalAlignment="Center"
             Content="This works, note the next property" Background="Violet"
             Height="{Binding $parent.Bounds.Height}"  Target="OrderInput"
             HorizontalContentAlignment="Center" Grid.Column="2" />
      <TextBox Grid.Column="3" Watermark="I do get focus" Name="OrderInput" />
</Grid>
```

### Expected behavior

Focus shifting when clicking anywhere in the bounding box

### Avalonia version

Avalonia 11.2.1

### OS

Windows

### Additional context

I have not tested other OSes, but I assume it might be affected there as well.
Perhaps changing `Background` on `Label` to default to `Transparent` would be a good fix here?

**Comment by timunie** [comment]
Yes, it needs a transparent background for hittest to work. That's by design. Updating the style however may be a good idea 🤔

**Comment by RivenSkaye** [comment]
Ah, makes sense. It was surprising though, having a block element with no initial hitbox when its entire purpose is making it easier to select another element with a smaller hitbox.

Will close this issue anyways, as this _is_ intended behavior then.

**Comment by timunie** [comment]
I meant it's intended that it needs a Background. So we may want to add it by default, might be an oversight when it was first created. Feel free to provide a PR.

**Comment by KieranDevvs** [comment]
@timunie Is this as simple as setting the background as transparent in the label constructor so that all label have a transparent background rather than none/null?

```diff
        /// <summary>
        /// Initializes instance of <see cref="Label"/> control
        /// </summary>
        public Label()
        {
+           Background = Brushes.Transparent;
        }
```

**Comment by timunie** [comment]
In the constructor would be a bad idea. Add it here as setter:

https://github.com/AvaloniaUI/Avalonia/blob/master/src%2FAvalonia.Themes.Fluent%2FControls%2FLabel.xaml#L5-L6

Need also to be updated for simple theme if you want to file a PR

**Comment by KieranDevvs** [comment]
> In the constructor would be a bad idea. Add it here as setter:
> 
> https://github.com/AvaloniaUI/Avalonia/blob/master/src%2FAvalonia.Themes.Fluent%2FControls%2FLabel.xaml#L5-L6
> 
> Need also to be updated for simple theme if you want to file a PR

Why would it be better to add the default behaviour to a theme rather than the base properties of the control itself?
I'm just thinking of cases where users don't want to use the default fluent theme and now they lose this standard default behaviour. Is it because it would be a potential breaking change to everyone rather than just people who have overridden the default themes?

**Comment by timunie** [comment]
In that case their theme should have it.

**Comment by KieranDevvs** [comment]
> In that case their theme should have it.

Perhaps that's the case but that doesn't help me understand why?
All I'm seeing is that this would violate polymorphic and DRY principles by forcing the implementer to add the behaviour rather than the base, thus if the behaviour changes, it needs to be made in multiple places. I'm sure there's a valid reason why it should be done this way, I'm just not currently seeing it?

**Comment by timunie** [comment]
Okay let me try to explain it a bit more. Setting it as you did will cause it to be a local set property which then can't be changed except for other local set values. So that will break all styles that have set it before. 

If anything we could consider is to override default metadata. Not sure if that's a breaking change or not.

See: https://docs.avaloniaui.net/docs/guides/custom-controls/how-to-create-advanced-custom-controls

https://docs.avaloniaui.net/docs/guides/styles-and-resources/setter-precedence#bindingpriority-values
